src/controllers/ratings.controller.js

```diff
@@ -0,0 +1,73 @@
+import db from '../database/models';
+import BaseRepository from '../repository/base.repository';
+import responseGenerator from '../helpers/responseGenerator';
+
+const { Ratings, Article } = db;
+
+/** Rating Class */
+class RatingsController {
+  /**
+   * Handles User ability to rate Articles
+   * @method
+   * @async
+   * @param  {Object}   req  - express http request object
+   * @param  {Object}   res  - express http response object
+   * @return {Object}        - returns an http response object
+   * @static
+   */
+  static async rateArticles(req, res) {
+    // needed parameters
+    const { id } = req.params;
+    const userId = req.currentUser.id;
+    const { ratings } = req.body;
+
+    const articleId = id;
+
+    try {
+      const article = await BaseRepository.findOneByField(Article, {
+        id: articleId
+      });
+
+      if (article) {
+        if (userId === article.authorId) {
+          return responseGenerator.sendError(
+            res,
+            400,
+            'You cannot rate your article'
+          );
+        }
+
+        const rater = await BaseRepository.findOneByField(Ratings, {
+          articleId,
+          userId
+        });
+
+        if (rater) {
```

Comment:
Nice implementation @codexempire ...you could also consider using the findOrCreate method which already exists on the baserep (not compulsory though). I love the thought process here :+1: